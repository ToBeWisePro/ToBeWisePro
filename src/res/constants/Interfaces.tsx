import {
  type BackButtonNavEnum,
  type NotificationsMenuOptionEnum,
} from "./Enums";

export interface RouteInterface {
  params: {
    currentQuotes: QuotationInterface[];
    editingQuote: QuotationInterface;
    quoteSearch: {
      filter: string;
      query: string;
    };
    editingExistingQuote: boolean;
    showBackButton: boolean;
    backButtonNavigationFunction: BackButtonNavEnum;
    notificationKey: string;
    notificationFilter: string;
  };
}

export interface NavigationInterface {
  addListener: any;
  // eslint-disable-next-line no-empty-pattern
  push: (ev: string, {}) => void;
  goBack: () => void;
  navigate: (e: string) => void;
}

export interface Subject {
  _id: number;
  value: string;
}

export interface QuotationInterface {
  _id?: number;
  quoteText: string;
  author: string;
  subjects: string;
  authorLink: string;
  videoLink: string;
  contributedBy: string;
  favorite: boolean;
  deleted: boolean;
}

export interface NotificationsMenuOptionProps {
  notificationsMenuOptionEnum: NotificationsMenuOptionEnum;
  label: string;
  state: any;
  setState?: any;
  toggleFunction?: any;
  bottomLine: boolean;
  errorMessage?: string | null;
}

export interface NavButtonInterface {
  buttonText: string;
  selected: boolean;
  navigationTarget: string;
  navigation: NavigationInterface;
}
