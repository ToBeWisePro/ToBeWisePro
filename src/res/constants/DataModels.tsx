

export const QuotationSchema = {
  name: "Quotation",
  properties: {
    _id: "int",
    quoteText: "string",
    author: "string",
    subjects: "string",
    authorLink: "string",
    videoLink: "string",
    contributedBy: "string?",
    favorite: "int",
  },
  primaryKey: "_id",
};

export const SubjectV1Schema = {
  name: "SubjectV1",
  properties: {
    _id: "int",
    value: "string",
  },
};

export const AuthorSchema = {
  name: "Author",
  properties: {
    _id: "int",
    subject: "string",
  },
};

