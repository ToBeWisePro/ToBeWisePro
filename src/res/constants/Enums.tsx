export enum Frequencies {
    "15 Minutes",
      "30 Minutes",
      "1 Hour",
      "2 Hours",
      "4 Hours",
      "6 Hours",
  }

  export enum BackButtonNavEnum {
    GoBack,
    ResetHomeScreen,
  }
  
  export enum QuoteContainerButtons {
    Add = "Add",
    Delete = "Delete",
    Edit = "Edit",
    Video = "Video",
    Favorite = "Favorite",
    Share = "Share",
  }

  export enum ContainerSize {
    Large,
    Small,
  }
  
  export enum NotificationsMenuOptionEnum {
    Toggle,
    TimeSelector,
    DBSelector,
    TextField,
    FrequencySelector
  }

  export enum IncludeInBottomNav {
    AutoScrollBar,
    PlayButton,
    Nothing,
  }