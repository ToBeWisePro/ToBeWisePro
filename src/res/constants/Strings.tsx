import LocalizedStrings from "react-localization";

export const strings = new LocalizedStrings({
    en: {
      screenName: {
        settings: "Settings",
        homeHorizontal: "HomeHorizontal",
        home: "Home",
        discover: "Discover",
        editQuote: "Edit Quotes",
        search: "Search Quotes",
        notificationsScreen: "Configure Notifications"
      },
      database: {
        safeChar: "%",
        defaultQuery: "Top 100",
        defaultFilter: "Subject",
      },
      filters: {
        author: "Author",
        subject: "Subject"
      },
      placeholderText: {
        author: "Author...",
        quote: "Quotation...",
        subjects: "Category 1, Category 2, ...",
        authorLink: "www.author.link",
        videoLink: "www.video.link",
      },
      labels: {
        author: "*Author",
        quote: "*Quotation",
        authorLink: "Link To Author",
        videoLink: "Link To Video",
        subjects: "*Subjects (comma separated)"
      },
      copy: {
        saveButton: "Save Changes",
        saveButtonBlank: "Add New Quote",
        editQuoteTitle: "Edit Quote",
        editQuoteTitleBlank: "Add Quote",
        shareMessage: "on ToBeWise",
        searchBarTitle: "Search",
        searchBarPlaceholderText: "Enter search query",
        errorMessages:{
          screenFailedToLoad: "Our developer isn't great, his bad code broke the app. Please quit and restart it"
        },
      },
      playPauseButtons: {
        play: "play-circle-outline",
        pause: "pause",
        slowDown: "fast-rewind",
        speedUp: "fast-forward"
      },
      customDiscoverHeaders:{
        all: "üåè Show All",
        addedByMe: "üôã Contributed By Me",
        deleted: "üóëÔ∏è Deleted",
        favorites: "‚ù§Ô∏è Favorite Quotations",
        top100: "üèÜ Top 100"
      },
      
      testText: "The AppCopy.tsx text works üí©",
      navbarHomeDefaultText: "Subject: üèÜ Top 100",
      navbarDiscoverDefaultText: "Discover",
    },
  
  });