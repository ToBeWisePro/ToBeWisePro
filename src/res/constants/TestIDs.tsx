export const TEST_IDS = {
  discoverScreen: "discover-screen",
  topNav: "top-nav",
  searchBar: "search-bar",
  alphabetListSection: "alphabet-list-section",
  bottomNav: "bottom-nav",
  dateTimePicker: "date-time-picker",
  iconHome: "icon-home",
  dropDownPicker: "dropdown-picker",
  selectorButton: "selector-button",
  PlayButtonBar: "play-pause-button",
  frequencySelector: "frequency-selector",
  clearButton: "clear-button",
  selectTimeButton: "select-time-button",
};
