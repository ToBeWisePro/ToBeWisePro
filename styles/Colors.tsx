export const LIGHT = '#FFFFFF'
export const DARK = '#242424'

export const PRIMARY_GREEN = '#459075'
export const PRIMARY_BLUE = '#4680B5'
export const PRIMARY_RED = '#FF0000'

export const GRAY_1 = '#333333'
export const GRAY_2 = '#4f4f4f'
export const GRAY_3 = '#828282'
export const GRAY_4 = '#BDBDBD'
export const GRAY_5 = '#E0E0E0'
export const GRAY_6 = '#F2F2F2'

export const GRADIENT_START = '#DFF3E3'
export const GRADIENT_END = '#80B2DF'
