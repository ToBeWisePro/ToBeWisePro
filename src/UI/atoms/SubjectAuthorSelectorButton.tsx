// The purpose of this code is to create a reusable button component which is used to switch between Subjects and Authors in the discover page
import React from "react";
import { TouchableWithoutFeedback } from "react-native-gesture-handler";
import { StyleSheet, View } from "react-native";
import { AppText } from "./AppText";
import { GRAY_3, LIGHT, PRIMARY_GREEN } from "../../../styles/colors";

interface Props {
  buttonText: string;
  selected: boolean;
  onPress: () => void;
}

export const SubjectAuthorSelectorButton: React.FC<Props> = ({
  buttonText,
  selected,
  onPress,
}: Props) => {
  return (
    <TouchableWithoutFeedback
      style={[
        styles.container,
        { backgroundColor: selected ? PRIMARY_GREEN : GRAY_3 },
      ]}
      onPress={onPress}
    >
      {selected ? (
        <View>
          <AppText style={styles.selected}>{buttonText}</AppText>
        </View>
      ) : (
        <View>
          <AppText style={styles.deselected}>{buttonText}</AppText>
        </View>
      )}
    </TouchableWithoutFeedback>
  );
};

const styles = StyleSheet.create({
  container: {
    // flex:1,
    justifyContent: "center",
    alignItems: "center",
    height: 32,
    width: 108,
    borderRadius: 5,
  },
  selected: {
    color: LIGHT,
  },
  deselected: {
    color: LIGHT,
  },
});
