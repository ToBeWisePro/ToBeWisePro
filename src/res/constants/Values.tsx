import {Dimensions} from "react-native";
import { globalStyles } from "../../../styles/GlobalStyles";

export const dbName: string = "QuotesDB"
export const largeQuoteContainerRefreshRate = 5000
export const autoScrollIntervalTime = 0
export const maxScrollIntervalTime = 50
export const defaultUsername = "Me"
export const maxWindowSize = Dimensions.get("window").height - ((globalStyles.navbar.height + globalStyles.navbar.playPauseHeight) - globalStyles.navbar.topHeight)