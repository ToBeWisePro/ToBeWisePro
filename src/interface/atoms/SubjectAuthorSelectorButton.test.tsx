import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';
import { SubjectAuthorSelectorButton } from './SubjectAuthorSelectorButton';
import '@testing-library/jest-native/extend-expect';
import { PRIMARY_GREEN, GRAY_3 } from '../../../styles/Colors';
import { TEST_IDS } from '../../res/constants/TestIDS';

describe('SubjectAuthorSelectorButton component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SubjectAuthorSelectorButton
        buttonText="Test Button"
        selected={true}
        onPress={jest.fn()}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onPress when button is pressed', () => {
    const mockOnPress = jest.fn();
    const { getByText } = render(
      <SubjectAuthorSelectorButton
        buttonText="Test Button"
        selected={true}
        onPress={mockOnPress}
      />
    );

    fireEvent.press(getByText('Test Button'));
    expect(mockOnPress).toHaveBeenCalled();
  });

  it('changes button background color based on selected prop', () => {
    const tree = renderer.create(
      <SubjectAuthorSelectorButton
        buttonText="Test Button"
        selected={true}
        onPress={jest.fn()}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();

    const updatedTree = renderer.create(
      <SubjectAuthorSelectorButton
        buttonText="Test Button"
        selected={false}
        onPress={jest.fn()}
      />
    ).toJSON();
    expect(updatedTree).toMatchSnapshot();
  });
});
