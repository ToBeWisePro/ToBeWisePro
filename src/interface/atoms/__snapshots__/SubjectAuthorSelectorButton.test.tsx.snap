// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubjectAuthorSelectorButton component changes button background color based on selected prop 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={3}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="selector-button"
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#459075",
        "borderRadius": 5,
        "height": 32,
        "justifyContent": "center",
        "width": 108,
      }
    }
  >
    <View>
      <Text
        ellipsizeMode="tail"
        maxFontSizeMultiplier={1}
        numberOfLines={9999}
        style={
          {
            "color": "#FFFFFF",
          }
        }
      >
        Test Button
      </Text>
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`SubjectAuthorSelectorButton component changes button background color based on selected prop 2`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={4}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="selector-button"
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#828282",
        "borderRadius": 5,
        "height": 32,
        "justifyContent": "center",
        "width": 108,
      }
    }
  >
    <View>
      <Text
        ellipsizeMode="tail"
        maxFontSizeMultiplier={1}
        numberOfLines={9999}
        style={
          {
            "color": "#FFFFFF",
          }
        }
      >
        Test Button
      </Text>
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`SubjectAuthorSelectorButton component renders correctly 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="selector-button"
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#459075",
        "borderRadius": 5,
        "height": 32,
        "justifyContent": "center",
        "width": 108,
      }
    }
  >
    <View>
      <Text
        ellipsizeMode="tail"
        maxFontSizeMultiplier={1}
        numberOfLines={9999}
        style={
          {
            "color": "#FFFFFF",
          }
        }
      >
        Test Button
      </Text>
    </View>
  </View>
</RNGestureHandlerButton>
`;
